andela/pippin-ah-backend
# Feature/163224026/twitter auth

#### What does this PR do?
- THIS PR sets up Twitter authentication for signing and logging users

#### Description of Task to be completed?
- [x] setup twitter authentication with Passport js
- [x] ensure a user receives a token upon signing up through twitter
- [x] ensure an existing user receives a token upon logging in through twitter

#### How should this be manually tested?
- set TWITTER_CONSUMER_KEY and TWITTER_CONSUMER_SECRET in the env file
- start the app
- navigate to `/api/v1/users/twitter`
- you should be redirected to Twitter for authentication
- login with a valid Twitter account or authorize app if already logged in
- server will respond with a JWT token and user details

#### What are the relevant pivotal tracker stories?
- [#163224026](https://www.pivotaltracker.com/story/show/163224026)
